c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -146,8 +146,11 @@ impl<'a, 'tcx> ExprRewriteVisitor<'a, 'tcx> {
                 let (pl, ref rv) = **x;
                 let pl_lty = self.acx.type_of(pl);
 
+                // FIXME: Needs changes to handle CELL pointers in struct fields.  Suppose `pl` is
+                // something like `*(_1.0)`, where the `.0` field is CELL.  This should be
+                // converted to a `Cell::get` call, but we would fail to enter this case because
+                // `_1` fails the `is_any_ptr()` check.
```

Comment:
should the check instead look for what the projection_ty is? would that allow the block to be entered at least, even if the rewrite is incorrect?

Reply:
It would need to look at the type of the outermost `Deref`, since that determines what location is actually being written to.  For example, in `*((*_1).0) = 3;`, we should look at the outermost `PointerId` on `(*_1).0`, and if it's `CELL`, rewrite to `(*_1).0.set(3);`.

Reply:
could we create issues from these FIXME's?

Reply:
Opened #924